contracts/tasks/gen-selector-library.ts

```diff
@@ -0,0 +1,73 @@
+import { task } from 'hardhat/config'
+import '@nomiclabs/hardhat-ethers'
+import * as fs from 'fs'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { selectors } from './lib'
+
+task('gen-selector-library', 'Generates a Solidity library with contract selectors for tests').setAction(
+    async (_args, hre: HardhatRuntimeEnvironment) => {
+        await hre.run('compile')
+
+        // ridiculously, this appears to be the only way to get hardhat to compile a specific subtree
+        // we are only updating the in-memory representation of the config, so it won't write this value out to disk
+        // be careful if you compose this task with other tasks in larger scripts!
+        hre.config.paths.sources = './test/mocks'
+        await hre.run('compile')
+
+        const contracts: string[] = [
+            'OwnershipFacet',
+            'DiamondCutFacet',
+            'DiamondLoupeFacet',
+            'GatewayGetterFacet',
+            'GatewayManagerFacet',
+            'GatewayMessengerFacet',
+            'CheckpointingFacet',
+            'TopDownFinalityFacet',
+            'XnetMessagingFacet',
+            'SubnetActorGetterFacet',
+            'SubnetActorManagerFacet',
+            'SubnetActorPauseFacet',
+            'SubnetActorRewardFacet',
+            'SubnetActorCheckpointingFacet',
+            'RegisterSubnetFacet',
+            'SubnetGetterFacet',
+            'SubnetActorMock',
+        ]
+
+        const resolveSelectors = async (contractName: string) => {
+            console.log(`Resolving selectors for ${contractName}...`)
+            const artifact = await hre.artifacts.readArtifact(contractName)
+            const iface = new hre.ethers.utils.Interface(artifact.abi)
+            const encodedSelectors = hre.ethers.utils.defaultAbiCoder
+                .encode(['bytes4[]'], [selectors({ interface: iface })])
+                .slice(2)
+            return [contractName, encodedSelectors]
+        }
+
+        const allSelectors = Object.fromEntries(await Promise.all(contracts.map(resolveSelectors)))
+
+        // Codegen.
+        let code = `// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.19;
```

Comment:
I think we use ^0.8.23 in most other contracts now. Is it worth updating here as well?